pkg/apis/awsprovider/v1beta1/awsproviderconfig_types.go

```diff
@@ -88,6 +88,9 @@ type AWSMachineProviderConfig struct {
 
 	// SpotMarketOptions allows users to configure instances to be run using AWS Spot instances.
 	SpotMarketOptions *SpotMarketOptions `json:"spotMarketOptions,omitempty"`
+
+	// Tenancy indicates if instance should run on shared or single-tenant hardware.
+	Tenancy string `json:"tenancy,omitempty"`
```

Comment:
Might be worth setting this as a typed string and having the valid options be part of this package as exported variables? WDYT? 

I would also suggest adding a comment to this to explain the valid and default options